### PR TITLE
replace gometalinter with golangci

### LINKED ISSUE #242: Change linter from gometalinter to golangci-lint
**Describe the bug**
gometalinter is being deprecated in favor of golangci-lint (see, https://github.com/alecthomas/gometalinter/issues/590).

We possibly need to move `make test` and the TravisCI job to the newer linter.

### PR DESCRIPTION
**Pull request by Madhu-1**
Fixes #242

### CODE REVIEW

**Comment by Madhu-1** [comment]
@ShyamsundarR @rootfs @gman0 PTAL

File: .travis.yml [hunk 1]
```diff
@@ -31,9 +31,9 @@ jobs:
         - pip install --user --upgrade pip
         - pip install --user yamllint
         # install gometalinter
```
Comment (ShyamsundarR) [inline .travis.yml]: (nit) comment needs a change

File: scripts/golangci.yml [hunk 2]
```diff
@@ -0,0 +1,152 @@
+---
+# This file contains all available configuration options
+# with their default values.
+
```
Comment (ShyamsundarR) [inline scripts/golangci.yml]: Request you mention the source for this file and if any options are changed here? Helps change things in the future or catch up to later lint configuration changes as they happen.

Comment (Madhu-1) [reply scripts/golangci.yml]: done

File: pkg/cephfs/nodeserver.go [hunk 3]
```diff
@@ -150,7 +150,7 @@ func (*NodeServer) mount(volOptions *volumeOptions, req *csi.NodeStageVolumeRequ
 
 	klog.V(4).Infof("cephfs: mounting volume %s with %s", volID, m.name())
 
-	if err = m.mount(stagingTargetPath, cr, volOptions, volID); err != nil {
```
Comment (rootfs) [inline pkg/cephfs/nodeserver.go]: so this is unused param is found by golangci?

File: scripts/golangci.yml [hunk 4]
```diff
@@ -0,0 +1,153 @@
+---
+# https://github.com/golangci/golangci-lint/blob/master/.golangci.example.yml
```
Comment (rootfs) [inline scripts/golangci.yml]: would it be possible to curl this file instead of checking it in?

File: pkg/cephfs/nodeserver.go (hunk 3 above)
Comment (Madhu-1) [reply pkg/cephfs/nodeserver.go]: yes it  found during static check by golangci

File: scripts/golangci.yml (hunk 4 above)
Comment (Madhu-1) [reply scripts/golangci.yml]: the reference is taken from the linked file, and I have modified some values as per our requirement.

**Pull request merged**